base/src/main/java/io/spine/validate/FieldContext.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validate;
+
+import com.google.common.base.Optional;
+import com.google.protobuf.Descriptors.FieldDescriptor;
+import io.spine.base.FieldPath;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.google.common.collect.Lists.newLinkedList;
+import static io.spine.util.Exceptions.newIllegalStateException;
+import static java.util.Collections.singleton;
+
+/**
+ * A context of a {@link FieldDescriptor}.
+ *
+ * <p>The context in essence is a list of parent field descriptors,
```

Comment:
`In particular, holds a hierarchy of field descriptors from the top-level descriptor to the descriptor of the current field.`

Also, please add some sample code to the `descriptors` field. It's not totally clear.